### PR TITLE
add build pool size parameter

### PR DESCRIPTION
**Pull request by smile37773**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 1]
```diff
@@ -77,6 +78,11 @@ def load_arguments(self, _):
                         "--app-insights-key or --app-insights, "
                         "will create a new Application Insights instance in the same resource group.",
                    validator=validate_tracing_parameters_asc_create)
+        c.argument('build_pool_size',
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_params.py:81]: add preview = true and note this is (enterprise only)

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 2]
```diff
@@ -102,6 +108,9 @@ def load_arguments(self, _):
                    deprecate_info=c.deprecate(target='az spring-cloud update --disable-app-insights',
                                               redirect='az spring-cloud app-insights update --disable',
                                               hide=True))
+        c.argument('build_pool_size',
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_params.py:112]: add preview = true and note this is (enterprise only)

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 3]
```diff
@@ -236,7 +245,7 @@ def prepare_logs_argument(c):
                 'main_entry', options_list=[
                     '--main-entry', '-m'], help="A string containing the path to the .NET executable relative to zip root.")
             c.argument(
-                'builder', help="The name of builder.", default="default")
+                'builder', help="The name of builder.", default="default", validator=validate_builder)
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_params.py]: add preview = true and note this is (enterprise only)

File: src/spring-cloud/azext_spring_cloud/_validators_enterprise.py [hunk 4]
```diff
@@ -121,3 +122,18 @@ def validate_buildpacks_binding_exist(cmd, namespace):
...
+    builder = client.build_service_builder.get(namespace.resource_group,
+                                               namespace.service,
+                                               DEFAULT_BUILD_SERVICE_NAME,
+                                               namespace.builder)
+    if builder.properties.provisioning_state != AppPlatformEnums.BuilderProvisioningState.SUCCEEDED:
+        raise CLIError('The provision state of builder {} is not succeeded, please choose a succeeded builder.'
+                       .format(namespace.builder))
+
+def validate_build_pool_size(namespace):
+    if namespace.sku.name != 'E0':
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_validators_enterprise.py]:
```python
from ._validator imports (_parse_sku_name)

if _parse_sku_name(namespace) == 'enterprise':
```

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 5]
```diff
@@ -150,6 +157,10 @@ def spring_cloud_update(cmd, client, resource_group, name, app_insights_key=None
     _update_application_insights_asc_update(cmd, resource_group, name, location,
                                             app_insights_key, app_insights, disable_app_insights, no_wait)
 
+    if build_pool_size is not None:
+        _update_default_build_agent_pool(
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/custom.py]: how about moving the `if build_pool_size is not None` inside `_update_default_build_agent_pool`

File: src/spring-cloud/azext_spring_cloud/_validators_enterprise.py (hunk 4 above)
Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/_validators_enterprise.py]: OK,updated

File: src/spring-cloud/azext_spring_cloud/_params.py (hunk 3 above)
Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/_params.py]: OK, added.

File: src/spring-cloud/azext_spring_cloud/_params.py (hunk 1 above)
Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/_params.py:81]: OK, added.

File: src/spring-cloud/azext_spring_cloud/_params.py (hunk 2 above)
Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/_params.py:112]: OK, added.

File: src/spring-cloud/azext_spring_cloud/custom.py (hunk 5 above)
Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/custom.py]: OK

**Review by yuwzho — APPROVED** [review]

**Pull request merged**